### PR TITLE
Add support for AssetLock and AssetUnlock transactions

### LINKED ISSUE #25: Asset Locks are not supported by Electrum
# Issue
AssetLocks and AssetUnlocks are not implemented in the codebase, and wallet fails to initialize connection with electrumx server

```
20241022T163959.981852Z |    ERROR | electrum.interface.[dash-electrum.pshenmic.dev:50002] | Exception in run: RequestCorrupted('cannot deserialize received transaction (txid 6364e6a49ff295d85d193dc4c76d476bfd7b2c4249ac41e837ce3047fa503a0c)')
```

https://explorer.dash.org/insight/tx/6364e6a49ff295d85d193dc4c76d476bfd7b2c4249ac41e837ce3047fa503a0c

### PR DESCRIPTION
**Pull request by LexxXell**
This PR implements support for AssetLock (type 8) and AssetUnlock (type 9) transactions, as specified in Dash's DIP-0027. It includes complete parsing and serialization of these transaction types, addressing issue #25.

Additional changes:
- Removed deprecated and unused components from dev-testnet to streamline codebase and improve maintainability.

Closes #25

### CODE REVIEW

File: electrum_dash/dash_tx.py [hunk 1]
```diff
@@ -698,156 +700,97 @@ def read_vds(cls, vds):
...
     @classmethod
     def read_vds(cls, vds):
-        return DashSubTxCloseAccount(
-            vds.read_uint16(),                          # version
-            vds.read_bytes(32),                         # regTxHash
-            vds.read_bytes(32),                         # hashPrevSubTx
-            vds.read_int64(),                           # creditFee
-            vds.read_bytes(96)                          # payloadSig
-        )
+        version = read_uint8(vds)                   # uint8_t
```
Comment (pshenmic) [inline electrum_dash/dash_tx.py:787]: You should document field names, not the types

File: electrum_dash/dash_tx.py [hunk 2]
```diff
@@ -698,156 +700,97 @@ def read_vds(cls, vds):
...
-                'creditFee: %s\n'
-                'newPubKey: %s\n'
-                % (self.version,
-                   bh2u(self.regTxHash[::-1]),
-                   bh2u(self.hashPrevSubTx[::-1]),
-                   self.creditFee,
-                   bh2u(self.newPubKey)))
+        outputs_str = '\n'.join(
+            ['  - Value: {}\n    ScriptPubKey: {}'.format(
+                co[0], bh2u(co[1])) for co in self.creditOutputs])
```
Comment (pshenmic) [inline electrum_dash/dash_tx.py:716]: What is the `co`?

File: electrum_dash/dash_tx.py [hunk 3]
```diff
@@ -698,156 +700,97 @@ def read_vds(cls, vds):
...
-
-class DashSubTxResetKey(ProTxBase):
-    '''Class representing DIP5 SubTxResetKey'''
-
-    __slots__ = ('version regTxHash hashPrevSubTx '
-                 'creditFee newPubKey payloadSig').split()
+    def __init__(self, version, count, creditOutputs):
+        self.version = version            # uint8_t
+        self.count = count                # uint8_t
+        self.creditOutputs = creditOutputs  # List of (value, scriptPubKey)
```
Comment (pshenmic) [inline electrum_dash/dash_tx.py:711]:
```suggestion
        self.creditOutputs = creditOutputs  # List of tx outpoint (value, scriptPubKey)
```

File: electrum_dash/transaction.py [hunk 4]
```diff
@@ -647,11 +647,11 @@ def read_vds(cls, vds, alone_data=False, tx=None) -> None:
             tx._version = header
 
         n_vin = vds.read_compact_size()
-        if n_vin < 1:
+        if n_vin < 1 and tx._tx_type not in (9,):
```
Comment (pshenmic) [inline electrum_dash/transaction.py:650]: Why do we need this check?

**Review by pshenmic — CHANGES_REQUESTED** [review]

File: electrum_dash/transaction.py (hunk 4 above)
Comment (LexxXell) [reply electrum_dash/transaction.py:650]: As I understand it, there may be no inputs for AssetUnlock. That's what the check is for.

File: electrum_dash/dash_tx.py (hunk 1 above)
Comment (LexxXell) [reply electrum_dash/dash_tx.py]: fixed

File: electrum_dash/dash_tx.py (hunk 2 above)
Comment (LexxXell) [reply electrum_dash/dash_tx.py]: co = credit_output. renamed. fixed

File: electrum_dash/dash_tx.py (hunk 3 above)
Comment (LexxXell) [reply electrum_dash/dash_tx.py]: fixed

**Review by pshenmic — APPROVED** [review]

**Pull request merged**